prime-router/src/main/kotlin/azure/SenderFilesFunction.kt

```diff
@@ -0,0 +1,256 @@
+package gov.cdc.prime.router.azure
+
+import com.fasterxml.jackson.module.kotlin.jacksonMapperBuilder
+import com.microsoft.azure.functions.HttpMethod
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import com.microsoft.azure.functions.annotation.AuthorizationLevel
+import com.microsoft.azure.functions.annotation.FunctionName
+import com.microsoft.azure.functions.annotation.HttpTrigger
+import gov.cdc.prime.router.ReportId
+import gov.cdc.prime.router.Sender
+import gov.cdc.prime.router.azure.db.tables.pojos.ReportFile
+import gov.cdc.prime.router.azure.db.tables.pojos.SenderItems
+import gov.cdc.prime.router.common.CsvUtilities
+import gov.cdc.prime.router.common.Hl7Utilities
+import gov.cdc.prime.router.messages.ReportFileMessage
+import gov.cdc.prime.router.tokens.OktaAuthentication
+import org.apache.logging.log4j.kotlin.Logging
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.util.UUID
+
+class SenderFilesFunction(
+    private val oktaAuthentication: OktaAuthentication = OktaAuthentication(PrincipalLevel.SYSTEM_ADMIN),
+    private val dbAccess: DatabaseAccess = DatabaseAccess(),
+    private val blobAccess: BlobAccess = BlobAccess()
+) : Logging {
+    /**
+     * The sender file end-point retrieves the reports that contributed to the specified output report.
+     */
+    @FunctionName("getSenderFiles")
+    fun getSenderFiles(
+        @HttpTrigger(
+            name = "getSenderFiles",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "sender-files"
+        ) request: HttpRequestMessage<String?>
+    ): HttpResponseMessage {
+        return oktaAuthentication.checkAccess(request) {
+            try {
+                val parameters = try {
+                    checkParameters(request)
+                } catch (e: IllegalArgumentException) {
+                    return@checkAccess HttpUtilities.badRequestResponse(request, e.message ?: "")
+                }
+                val (status, payload) = processRequest(parameters)
+                when (status) {
+                    Status.OK -> HttpUtilities.okResponse(request, payload)
+                    Status.BAD_REQUEST -> HttpUtilities.badRequestResponse(request, payload)
+                    Status.NOT_FOUND -> HttpUtilities.notFoundResponse(request, payload)
+                }
+            } catch (e: Exception) {
+                logger.error("Internal error: $e")
+                HttpUtilities.internalErrorResponse(request)
+            }
+        }
+    }
+
+    data class FunctionParameters(
```

Comment:
Using a common data class to pass info down instead of multiple function parameters.